pkg/router/template/router.go

```diff
@@ -450,6 +451,8 @@ func (r *templateRouter) AddRoute(id string, route *routeapi.Route, host string)
 		}
 	}
 
+	config.RoutingKeyName = fmt.Sprintf("%x", md5.Sum([]byte(fmt.Sprintf("%s %s", config.TLSTermination, backendKey))))
```

Comment:
3 function calls and a cast on one line? seems a bit much.
I'll leave it to @ramr or @marun to tell me if just `config.TLSTermination` and `backendKey` are unique/adequate.